batavia/types/Set.js

```diff
@@ -7,7 +7,20 @@ batavia.types.Set = function() {
     function Set(args, kwargs) {
         Object.call(this);
         if (args) {
-            this.update(args);
+            // Fast-path for native Array objects.
+            if (batavia.isArray(args)) {
+                for (var i = 0; i < args.length; i++) {
+                    this[args[i]] = args[i];
+                }
+            } else if (batavia.isinstance(args, [batavia.types.FrozenSet, batavia.types.List, batavia.types.Set, batavia.types.Str, batavia.types.Tuple])) {
+                var iterobj = batavia.builtins.iter([args], null);
+                var self = this;
+                batavia.iter_for_each(iterobj, function(val) {
+                    self[val] = val;
+                });
+            } else {
+                throw new batavia.builtins.TypeError("'" + batavia.type_name(args) + "' object is not iterable");
+            }
```

Comment:
That makes sense for FrozenSet, but this is in Set, which *does* have an update method (and in this case, an update method that now has obvious flaws, because it doesn't handle a bunch of use cases).